imagePNG.ml

```diff
@@ -1222,14 +1230,17 @@ let write_png fn img =
             add_byte (a land mask));
         done
       done
-    | _ -> Printf.fprintf stderr "%i\n%!" bd; assert false
+    | _ -> if (!debug) then Printf.fprintf stderr "bd: %i\n%!" bd;
+      raise (Corrupted_image "PNG: img.pixels * bd (bitdepth) mismatch")
   );
   let data = Buffer.contents buf in
   let data = compress_string data in
 
   let datalen = String.length data in
-  let max_idat_len = 1048576 in (* 2^20 sould be enough *)
-  (* FIXME constant too big for 32 bit architectures...*)
+  let max_idat_len = 1048576 in (* 2^20 should be enough *)
+  (* FIXME constant too big for 32 bit architectures...
+     maybe Sys.max_string_length
```

Comment:
Yes, why not. This would solve the problem on 64 bit architectures, right?

Reply:
Yep, but I didn't investigate which size would be the max, ie what the underlying code layers expect to be able to do with this. Ideally we'll have progressive rendering at some point to avoid all these limits.

Reply:
Yeah, that would be better.

Reply:
It's too late now, but for what it's worth given that `( ** )` is part of `Pervasives`, it would be much clearer to have used `2**20`.

Reply:
`( ** )` is not great, because it is not as efficient for powers of two.

Reply:
It's a constant. Does performance really matter so urgently for a constant number?

Reply:
If you're really complaining about performance of *a constant*, then use `1 lsl 20`, not a magic number than has to be explained in a comment.

Reply:
Oh right, sorry. I was thinking at the `pow_of_2` thing again. Of course, I don't care about constants!